File: Server/src/main/java/com/ashafee/ccserver/challenge/Challenge.java
package com.ashafee.ccserver.challenge;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity @Data @NoArgsConstructor
public class Challenge {
    @Id @GeneratedValue(strategy= GenerationType.AUTO)
    private long challengeID;
    private String name;
    private Language language;
    private String initialCode;
    private String description;
    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "challengeID") //extend by instead referencing a list of challenge-test from another table
    private Set<ChallengeTest> tests = new HashSet<>(); //that is, challenge independent tests
}   //Extension: last modified date, author

File: UserClient/ClientModel/Challenge.cs
namespace ClientModels
{
    public enum Language { JAVA }

    public class Challenge : IChallenge
    {
        public Challenge() { }

        public Challenge(long challengeID, string name, Language language, string initialCode, string description)
        {
            this.challengeID = challengeID;
            this.name = name;
            this.language = language;
            this.initialCode = initialCode;
            this.description = description;
        }

        public long challengeID { get; set; }

        public string name { get; set; }

        public Language language { get; set; }

        public string initialCode { get; set; }

        public string description { get; set; }
    }
}